pkg/controllers/provisioning/provisioner.go

```diff
@@ -162,6 +162,7 @@ func (p *Provisioner) GetPendingPods(ctx context.Context) ([]*corev1.Pod, error)
 	pods = lo.Reject(pods, func(po *corev1.Pod, _ int) bool {
 		if err := p.Validate(ctx, po); err != nil {
 			log.FromContext(ctx).WithValues("Pod", klog.KRef(po.Namespace, po.Name)).V(1).Info(fmt.Sprintf("ignoring pod, %s", err))
+			metrics.IgnoredPodTotal.Inc()
```

Comment:
This is going to be a counter, but I don't think we want it to be a counter. I think we want this to be a gauge that tracks the current number of ignored pods at any given point in time.